### PR TITLE
Configurable installation

### PR DESCRIPTION
**Pull request by scoopex**
Added the possibility to modify the installation location and profile change behavior to have the possibility make pulumi ready for automated installations.

- configurable install location: set PULUMI_INSTALL to define custom location
- configurable profile modification
  - set PULUMI_PROFILE_FILE="no" to skip profile modification
  - set PULUMI_PROFILE_FILE="<path>" to use another file to add the pulumi path

### CODE REVIEW

**Comment by jaxxstorm** [comment]
Hi @scoopex - thanks for this PR! It inspired me to land #77 and I'd like to make your changes compatible with that. Do you mind if I fork from your branch and add some tests?

**Comment by scoopex** [comment]
feel free :-)

**Comment by XBeg9** [comment]
@jaxxstorm do you have any timeline for this? It's hard to configure custom cache directories on CI, especially if we are using `/mnt/ramdisk`.

**Comment by jaxxstorm** [comment]
Hey all, I haven't forgotten about this and hoping to dedicate some time and finish it shortly, sorry for the delay!

**Comment by XBeg9** [comment]
Any updates?

File: dist/install.sh [hunk 1]
```diff
@@ -117,15 +117,20 @@ say_white "+ Downloading ${TARBALL_URL}..."
 
 TARBALL_DEST=$(mktemp -t pulumi.tar.gz.XXXXXXXXXX)
 
+if [ -z "$PULUMI_INSTALL" ];then
+   PULUMI_INSTALL="$HOME/.pulumi"
+fi
```
Comment (AaronFriel) [inline dist/install.sh:122]: Perhaps we should use `PULUMI_HOME` for this variable instead? PULUMI_HOME defaults to `~/.pulumi` and if the user has this set, it increases the likelihood that the subsequent install will work correctly.

Per: https://www.pulumi.com/docs/reference/cli/environment-variables/

**Pull request closed without merging**